src/Microsoft.Unity.Analyzers/IndirectionMessage.cs

```diff
@@ -0,0 +1,110 @@
+/*--------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *-------------------------------------------------------------------------------------------*/
+
+//using System.Collections.Immutable;
+//using System.Threading.Tasks;
+//using Microsoft.CodeAnalysis;
+//using Microsoft.CodeAnalysis.CodeFixes;
+//using Microsoft.CodeAnalysis.CSharp;
+//using Microsoft.CodeAnalysis.CSharp.Syntax;
+//using Microsoft.CodeAnalysis.Diagnostics;
+//using Microsoft.Unity.Analyzers.Resources;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Unity.Analyzers.Resources;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class IndirectionMessageAnalyzer : DiagnosticAnalyzer
+	{
+		internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			id: "UNT0019",
+			title: Strings.IndirectionMessageDiagnosticTitle,
+			messageFormat: Strings.IndirectionMessageDiagnosticMessageFormat,
+			category: DiagnosticCategory.Performance,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
+			description: Strings.IndirectionMessageDiagnosticDescription);
+
+		internal const string UpdateId = "UNT0019";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzeExpression, SyntaxKind.SimpleMemberAccessExpression);
+		}
+
+
+		private static void AnalyzeExpression(SyntaxNodeAnalysisContext context)
+		{
+			if (!(context.Node is MemberAccessExpressionSyntax))
+				return;
+
+			var access = (MemberAccessExpressionSyntax)context.Node;
+			var symbol = context.SemanticModel.GetSymbolInfo(access);
+
+			if (symbol.Symbol == null)
+				return;
+
+			if (!(symbol.Symbol is IPropertySymbol property))
```

Comment:
property is unused after so no need to use the full pattern matching syntax:

```csharp
if (!(symbol.Symbol is IPropertySymbol))
				return;
```

should be enough?